weed/command/filer_backup.go

```diff
@@ -123,6 +126,21 @@ func doFilerBackup(grpcDialOption grpc.DialOption, backupOption *FilerBackupOpti
 		return setOffset(grpcDialOption, sourceFiler, BackupKeyPrefix, int32(sinkId), lastTsNs)
 	})
 
+	if dataSink.IsIncremental() && *filerBackupOptions.retentionDuration > 0 {
+		go func() {
+			for {
+				now := time.Now()
+				next := now.Add(time.Hour * 24)
+				next = time.Date(next.Year(), next.Month(), next.Day(), 0, 0, 0, 0, next.Location())
+				t := time.NewTimer(next.Sub(now))
+				<-t.C
+				key := util.Join(targetPath, now.Add(-1*time.Hour*24*time.Duration(*filerBackupOptions.retentionDuration)).Format("2006-01-02"))
```

Comment:
If this backup has not been run for a few days, this logic would only delete the last folder.

Reply:
Normal backups need to be run continuously, otherwise they should be deleted manually.